mbed_targets/_internal/target_attribute_hierarchy_parsers/accumulating_attribute_parser.py

```diff
@@ -13,18 +13,31 @@
 from typing import Dict, List, Any, Deque
 
 ACCUMULATING_ATTRIBUTES = ["extra_labels", "macros", "device_has", "features", "components"]
-ALL_ACCUMULATING_ATTRIBUTES = ACCUMULATING_ATTRIBUTES + [
-    "extra_labels_remove",
-    "extra_labels_add",
-    "macros_remove",
-    "macros_add",
-    "device_has_remove",
-    "device_has_add",
-    "features_remove",
-    "features_add",
-    "components_remove",
-    "components_add",
-]
+ACCUMULATING_ATTRIBUTES_MODIFIERS = {"add": "_add", "remove": "_remove"}
+
+
+def _modifying_attribute_name(attribute: str, modifying_action: str) -> str:
```

Comment:
Should this be `modify_attribute_name`? Or is a "modifying attribute" a thing? We seem to pass an "accumulating attribute" to this. TBH I'm finding some of the terminology confusing so if I'm making no sense ignore me.

Reply:
No its true, I'm using modifying here as an adjective and not a verb, as in you get an accumulating attribute (eg. "extra_labels") and then you get attributes that modify that accumulating attribute (eg. "extra_labels_add", "extra_labels_remove"). But yeah its confusing.  I'll have a think about better naming.

Reply:
I have spent hours going round in circles about naming. I have added a bit more explanation to the docstring in the hope that is helps a little.